cirq-google/cirq_google/serialization/serializable_gate_set.py

```diff
@@ -24,10 +24,11 @@
 _GateOrFrozenCircuitTypes = Union[Type[cirq.Gate], Type[cirq.FrozenCircuit]]
 
 
-class SerializableGateSet(serializer.Serializer):
-    """A class for serializing and deserializing programs and operations.
+class _SerializableGateSet(serializer.Serializer):
```

Comment:
Should we deprecate the functions in this class?  If we deprecate the functions in the class, then any use will generate a warning, right?  That way, instantiation won't generate a warning, but if it is used to serialize a class, it will generate a warning.

Reply:
Hmmm yeah potentially. One concern is that there could be more call sites to the methods than the constructor since some code interacts with SerializableGateSet via the `Serializer` interface, leading to more tests that need to be updated. More work, but certainly less hacky.

Reply:
I think it would be preferable to at least deprecate the serialize() method.  That way people get a message when they use the most common call site at least.

Reply:
Chatted offline - the most common call site for users is via global gatesets, which have a separate deprecation warning in place to tell users about SerializableGateSet deprecation.